module/zfs/dsl_deadlist.c

```diff
@@ -810,6 +822,27 @@ dsl_deadlist_insert_bpobj(dsl_deadlist_t *dl, uint64_t obj, uint64_t birth,
 	dle_enqueue_subobj(dl, dle, obj, tx);
 }
 
+/*
+ * Prefetch metadata required for dsl_deadlist_insert_bpobj().
+ */
+static void
+dsl_deadlist_prefetch_bpobj(dsl_deadlist_t *dl, uint64_t obj, uint64_t birth)
+{
+	dsl_deadlist_entry_t dle_tofind;
+	dsl_deadlist_entry_t *dle;
+	avl_index_t where;
+
+	ASSERT(MUTEX_HELD(&dl->dl_lock));
+
+	dsl_deadlist_load_tree(dl);
+
+	dle_tofind.dle_mintxg = birth;
+	dle = avl_find(&dl->dl_tree, &dle_tofind, &where);
+	if (dle == NULL)
+		dle = avl_nearest(&dl->dl_tree, where, AVL_BEFORE);
```

Comment:
How about `ASSERT(dle, !=, NULL)` here, or move `dle_prefetch_subobj()` under a `!= NULL check`.

Reply:
It is a copy of dsl_deadlist_insert_bpobj().  I don't see much sense to assert it, since it would crash very soon on the NULL dereference.  Condition here would look confusing, like if it is possible.  And this is a very old code, that seems to be doing fine as is.

Reply:
The value I see in an ASSERT is that it makes it clear the author believes a NULL is impossible here.  And it's not simply that they overlooked this case.  But I don't feel strongly about it.